src/Integration.Vsix/SonarLintTracker/SnapshotFactory.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace SonarLintTracker
+{
+    class SnapshotFactory : TableEntriesSnapshotFactoryBase
+    {
+        public IssuesSnapshot CurrentSnapshot { get; private set; }
+
+        public SnapshotFactory(IssuesSnapshot snapshot)
+        {
+            this.CurrentSnapshot = snapshot;
+        }
+
+        internal void UpdateMarkers(IssuesSnapshot snapshot)
+        {
+            this.CurrentSnapshot.NextSnapshot = snapshot;
+            this.CurrentSnapshot = snapshot;
+        }
+
+        #region ITableEntriesSnapshotFactory members
+        public override int CurrentVersionNumber
+        {
+            get
+            {
+                return this.CurrentSnapshot.VersionNumber;
+            }
+        }
+
+        public override void Dispose()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1186)